src/nvim/os/stdpaths.c

```diff
@@ -93,6 +96,48 @@ bool appname_is_valid(void)
   return true;
 }
 
+/// Remove duplicate directories in the given string.
+/// e.g, "/usr/local/share:/usr/share:/usr/share"
+/// to "/usr/local/share:/usr/share"
+/// @param[in]  List of directories possibly with duplicates
+/// @param[out]  List of directories without duplicates
+static char *xdg_remove_duplicate(char *ret)
+{
+  CharVec data = KV_INITIAL_VALUE;
+  char *saveptr;
+
+  char *token = os_strtok(ret, ":", &saveptr);
+  while (token != NULL) {
+    // Check if the directory is not already in the list
+    bool is_duplicate = false;
+    for (size_t i = 0; i < kv_size(data); i++) {
+      if (strcmp(kv_A(data, i), token) == 0) {
+        is_duplicate = true;
+        break;
+      }
+    }
+    // If it's not a duplicate, add it to the list
+    if (!is_duplicate) {
+      kv_push(data, xstrdup(token));
```

Comment:
I don't think the `xstrdup()` is necessary. I think the existing token will still remain valid even as `os_strtok()` returns the next.

Reply:
okay thx.